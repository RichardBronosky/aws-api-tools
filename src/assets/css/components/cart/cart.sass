@import '../../mixins'
@import '../../variables'

.cart-page
  margin-top: 1em
  .bold
    font-weight: 600
  h1
    color: $dark-grey
    line-height: 1
    font-size: $txt-xlarge
    font-weight: 300
    padding: 0
    text-align: left
    margin-bottom: 1.2rem
    margin-top: 1.5rem
    +medium-down
      text-align: center
  .cart-product-list-container
    margin-right: 1em
    +medium-down
      margin-top: 0
    .financing-disclaimer
      color: $red
      text-transform: uppercase
    .slider
      margin-top: 1.5em
      +large-up
        margin-top: 0
      h2 
        font-size: $txt-medium
      .slick-slider
        +large-up
          margin-left: -2em
          margin-right: -2em
        .buy-now-popup
          .buy-now
            font-size: $txt-xsmall
  
  .checkout-sticky
    .order-detail-expander
      padding: 0

  .checkout-container
    position: fixed
    bottom: 0
    width: 100%!important
    z-index: 1
    border-top: 2px solid $grey
    margin-left: 0 !important
    +large-up
      margin-top: 0 !important
      position: -webkit-sticky
      position: sticky
      top: 10.375em!important
      right: 0
      height: 100%
      width: 23%!important
      z-index: 8
      border: 0
    .checkout-sticky
      margin-top: 0
.cart-bottom-checkout
  margin: 1rem 0 .2rem 0
  background-color: $primary-invert
  .checkout-sticky
    .checkout-info
      &.bottom
        margin: 0
  .button-container
    display: flex
    .checkout-btn-container
      width: 100% !important
    .paypal-btn-container
      margin-top: .5rem
      padding: 0
      +medium-up
        display: none!important
      .paypal
        padding: 0 1rem
        .loader
          height: 2rem
          width: 100%
          +medium-down
            width: 33.75%
