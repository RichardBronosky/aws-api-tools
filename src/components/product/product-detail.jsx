import React from 'react'
import RTGLink from '../shared/link'
import { InView } from 'react-intersection-observer'
import { trackImpression } from '../../lib/helpers/google-tag-manager'
import { decodeHtml } from '../../lib/helpers/string-helper'
import { getRegionSkuList, productAvailability } from '../../lib/helpers/product'
import Banner from '../shared/banner'
import Breadcrumb from '../shared/breadcrumbs'
import SimpleSlider from '../shared/slider'
import ProductInfo from './product-info'
import ProductTile from './product-tile'
import ProductJsonLd from './product-json-ld'
import classNames from 'classnames'
import { getRegionZone } from '../../lib/helpers/geo-location'
import shopIcon from '../../assets/images/shopIcon.png'
import ProductTitlePricing from './product-parts/product-title-pricing'
import ProductPromotions from './product-parts/product-promotions'
import RecentlyViewed from '../shared/recently-viewed'
import { connect } from 'react-redux'
import SaleFlag from '../shared/sale-flag'
import ProductIncludesSlider from './product-parts/product-includes-slider'
import ProductDescription from './product-parts/product-description'
import { isMobile } from 'react-device-detect'

class ProductDetail extends React.Component {
  constructor(props) {
    super(props)
    this.myRef = React.createRef()
  }

  createCrumb = (crumb, slugPath) => {
    return {
      altDesc: `View All ${ crumb } Products`,
      id: slugPath,
      slug: slugPath,
      text: crumb,
    }
  }

  generatePDPCrumbs = product => {
    let crumbs = []
    if (product.breadcrumb && product.breadcrumb.breadcrumb_label && product.breadcrumb.breadcrumb_url) {
      crumbs.push(this.createCrumb(product.breadcrumb.breadcrumb_label, product.breadcrumb.breadcrumb_url))
    }
    if (product.title) {
      crumbs.push(this.createCrumb(decodeHtml(product.title), window.location.pathname))
    }
    return {
      crumbs: crumbs,
    }
  }

  scrollToBonusBuys = () => window.scrollTo(0, this.myRef.current.offsetTop - 180)

  render() {
    const { isMobile, product, room, see_in_room, banners } = this.props
    if (product && typeof product !== typeof undefined) {
      const description = product.description
        ? product.description.replace(
            'Assembly required.',
            `<p class="uppercase"><strong>Customer assembly required.</strong></p>`
          )
        : ''
      const region = getRegionZone().region
      const zone = getRegionZone().zone
      let items_in_room = getRegionSkuList(product.items_in_room, region)
      const also_in_collection = getRegionSkuList(product.also_in_collection, region)
      const available_complete_your_room_items = product.complete_your_room_items
        ? product.complete_your_room_items.filter(p => productAvailability(p))
        : null
      const available_you_may_also_like_items = product.you_may_also_like_items
        ? product.you_may_also_like_items.filter(p => productAvailability(p))
        : null
      let promotions
      if (product && product.promotions) {
        promotions = product.promotions[`${ region }_${ zone }`] || product.promotions[`${ region }_0`]
      }
      return (
        <>
          <div
            className={ classNames('product-details', {
              room: room,
            }) }
          >
            <div className="grid-x grid-margin-x ">
              <div className="cell small-12 medium-6">
                <Breadcrumb data={ this.generatePDPCrumbs(product) } includeHeading={ true } />
              </div>

              <div className="cell small-12 medium-6">
                { see_in_room && !product.single_item_room && (
                  <RTGLink
                    className="product-shop-link large-right"
                    data={ {
                      slug: see_in_room.link,
                      title: 'Shop the Room and Save',
                      category: 'pdp',
                      action: 'shop the room and save click',
                      label: 'shop-and-save',
                    } }
                    key={ see_in_room.sku }
                  >
                    <img className="shopIcon" src={ shopIcon } alt="shop-icon" />
                    shop the room &#38; save >
                  </RTGLink>
                ) }
              </div>
            </div>
            <div className="product-details-card grid-margin-y">
              { promotions && (
                <SaleFlag className="promo-flag" onClick={ this.scrollToBonusBuys }>
                  <span role="button" className="ribbon">
                    CLICK FOR BONUS OFFER!
                  </span>
                </SaleFlag>
              ) }
              <div className="cell">
                { product && <ProductInfo data={ product } viewType="list" fullWidth promotions={ promotions } /> }
              </div>
            </div>
            { items_in_room && items_in_room.length > 0 && !product.single_item_room && (
              <div className="grid-container">
                <ProductIncludesSlider items_in_room={ items_in_room } heading="Items in this room" />
              </div>
            ) }

            <ProductDescription product={ product } items_in_room={ items_in_room } />

            { promotions && (
              <div ref={ this.myRef } className="grid-x grid-margin-y grid-margin-x">
                <ProductPromotions
                  product={ product }
                  promotions={ promotions }
                  scrollToBonusBuys={ this.scrollToBonusBuys }
                  promoType={ promotions.offer_template }
                />
              </div>
            ) }
            { also_in_collection && also_in_collection.length > 0 && (
              <div className="grid-container">
                <div className="product-collection-slider grid-x grid-y grid-margin-x grid-margin-y grid-padding-y">
                  <div className="cell small-12">
                    <SimpleSlider
                      data={ {
                        heading: 'ALSO IN THIS COLLECTION',
                      } }
                    >
                      { also_in_collection.map((item, index) => (
                        <ProductTile
                          sku={ item.sku }
                          key={ `collection_items_${ item.sku }` }
                          viewType="grid"
                          index={ index }
                        />
                      )) }
                    </SimpleSlider>
                  </div>
                </div>
              </div>
            ) }
            { available_complete_your_room_items && available_complete_your_room_items.length > 0 && (
              <div className="grid-container">
                <div className="product-collection-slider grid-x grid-y grid-margin-x grid-margin-y grid-padding-y">
                  <div className="cell small-12">
                    <SimpleSlider
                      data={ {
                        heading: 'Complete Your Room',
                      } }
                    >
                      { available_complete_your_room_items.map((item, index) => (
                        <ProductTile
                          sku={ item.sku }
                          key={ `available_complete_your_room_items_${ item.sku }` }
                          viewType="grid"
                          index={ index }
                        />
                      )) }
                    </SimpleSlider>
                  </div>
                </div>
              </div>
            ) }
            { see_in_room && !product.single_item_room && product.see_in_room && (
              <div className="product-in-room cell small-12 grid-x grid-margin-y grid-padding-y">
                <div className="cell small-12">
                  <div className="cell small-12 header">
                    <h2>SEE THE ROOM</h2>
                  </div>
                  <div className="cell small-12 view-all-link">
                    <RTGLink
                      className=" "
                      data={ {
                        heading: 'SEE IN A ROOM',
                        slug: see_in_room.link,
                        text: 'BUY THE ROOM & SAVE >',
                      } }
                    />
                  </div>
                  <RTGLink
                    data={ {
                      slug: see_in_room.link,
                      title: 'See in Room',
                      category: 'pdp',
                      action: 'see in room click',
                      label: 'see-in-room',
                    } }
                  >
                    <img className="see-in-room-slider-img" src={ see_in_room.image } alt={ see_in_room.product.title } />
                  </RTGLink>
                  <ProductTitlePricing product={ see_in_room.product } moreInfo />
                </div>
              </div>
            ) }
            { available_you_may_also_like_items && available_you_may_also_like_items.length > 0 && (
              <div className="product-collection-slider grid-x grid-y grid-margin-x grid-margin-y grid-padding-y">
                <div className="cell small-12 grid-x grid-margin-x">
                  <SimpleSlider
                    data={ {
                      heading: 'You May Also Like',
                    } }
                  >
                    { available_you_may_also_like_items.map((item, index) => (
                      <ProductTile
                        sku={ item.sku }
                        key={ `available_you_may_also_like_items_${ item.sku }` }
                        viewType="grid"
                        index={ index }
                      />
                    )) }
                  </SimpleSlider>
                </div>
              </div>
            ) }
            { banners && banners.bannerPlacementMiddle && (
              <InView
                as="div"
                onChange={ inView => trackImpression(inView, 1, banners.bannerPlacementMiddle) }
                className="product-middle-banner"
                triggerOnce={ true }
              >
                <Banner data={ banners.bannerPlacementMiddle } />
              </InView>
            ) }
            <InView
              onChange={ inView => trackImpression(inView, 2, { title: 'product recently viewed' }) }
              triggerOnce={ true }
            >
              <RecentlyViewed sku={ product.sku } />
            </InView>
          </div>
          { banners && banners.bannerFinancingOptions && (
            <InView
              as="div"
              onChange={ inView => trackImpression(inView, 3, banners.bannerPlacementMiddle) }
              className="product-finance-banner"
              triggerOnce={ true }
            >
              <Banner data={ banners.bannerFinancingOptions } />
            </InView>
          ) }
          <ProductJsonLd product={ product } />
        </>
      )
    } else {
      return null
    }
  }
}

const mapStateToProps = state => {
  return { isMobile: state.global.isMobile }
}

export default connect(mapStateToProps)(ProductDetail)
