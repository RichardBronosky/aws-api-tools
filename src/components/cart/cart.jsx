import React from 'react'
import classNames from 'classnames'
import CondensedContact from '../shared/condensed-contact-links'
import CartProduct from './cart-parts/cart-product'
import CreditCardBanner from './cart-parts/credit-card-banner'
import CartEmpty from './cart-parts/cart-empty'
import CheckoutSticky from '../checkout/checkout-sticky/checkout-sticky'
import PaymentMethods from './cart-parts/payment-methods'
import PromotionContentGroup from '../shared/promotion-content-group'
import { removeUnavailableItems } from '../../lib/helpers/cart'
import '../../assets/css/components/cart/cart.sass'

export default ({
  cart,
  isMobile,
  onAddActiveAddon,
  onRemoveActiveAddon,
  skusNotAvailable,
  discount,
  showPayPal,
  promoTargetSkus,
  setCartState,
}) => (
  <div className="cart-page">
    <h1 id="main-cart-title" tabIndex="-1">
      CART
    </h1>
    <div
      className={ classNames('grid-x', {
        'grid-margin-x': isMobile,
      }) }
    >
      <div
        className={ classNames('cart-product-list-container small-12 medium-12 large-9 grid-x grid-padding-y', {
          cell: isMobile,
        }) }
      >
        { cart &&
          cart.cartItems &&
          cart.cartItems.map((item, index) => (
            <CartProduct
              key={ index }
              product={ item.product }
              index={ index }
              productCount={ cart.cartItems.length }
              quantity={ item.quantity }
              price={ item.price }
              unavailableItem={ skusNotAvailable.filter(prod => prod.sku === item.sku)[0] }
              activeAddons={ item.activeAddons }
              onAddActiveAddon={ onAddActiveAddon }
              onRemoveActiveAddon={ onRemoveActiveAddon }
            />
          )) }
        { (!cart || (cart && cart.cartItems && cart.cartItems.length < 1) || (cart && !cart.cartItems)) && <CartEmpty /> }
        { cart && cart.cartItems && cart.cartItems.length > 0 && (
          <>
            <PromotionContentGroup targetSkus={ promoTargetSkus } isCart />
            <div className="cart-bottom-checkout cell small-12 grid-x">
              <div className="cell large-8 payment-methods">
                <PaymentMethods />
              </div>
              <CheckoutSticky
                cart={ cart }
                productsAvailable={ skusNotAvailable.length < 1 }
                removeUnavailableItems={ () => removeUnavailableItems(setCartState, skusNotAvailable) }
                onlyButtons
                discount={ discount }
                showPayPal={ showPayPal }
              />
            </div>
          </>
        ) }
        { cart && cart.cartItems && cart.cartItems.length > 0 && (
          <>
            <p className="financing-disclaimer cell small-12">*when paying with a rooms to go credit card</p>
            <CreditCardBanner />
          </>
        ) }
        <CondensedContact />
      </div>
      <div className="checkout-container cell small-12 large-2 grid-margin-y">
        { cart && cart.cartItems && cart.cartItems.length > 0 && (
          <CheckoutSticky
            cart={ cart }
            isMobile={ isMobile }
            productsAvailable={ skusNotAvailable.length < 1 }
            removeUnavailableItems={ () => removeUnavailableItems(setCartState, skusNotAvailable) }
            rightSticky
            discount={ discount }
            showPayPal={ showPayPal }
          />
        ) }
      </div>
    </div>
  </div>
)
