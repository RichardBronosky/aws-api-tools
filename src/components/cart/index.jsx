import React from 'react'
import { connect } from 'react-redux'
import { getLastParamOfPath } from '../../lib/helpers/string-helper'
import { verifyAndUpdateCart, cartUpdate, getPromoTargetSkus, loadCartFromOrder } from '../../lib/helpers/cart'
import { setupAnalytics } from '../../lib/helpers/google-tag-manager'
import Cart from './cart'

let apiCalled = false

class CartWrapper extends React.Component {
  state = {
    skusNotAvailable: [],
    discount: null,
    showPayPal: false,
  }

  setCartState = state => this.setState({ ...this.state, ...state })

  setApiCalled = called => {
    apiCalled = called
  }

  componentDidMount() {
    const { cart } = this.props
    const { discount, skusNotAvailable } = this.state
    setupAnalytics({ pageData: { type: 'cart', title: 'Cart', path: '/cart' } })
    const lastParam = getLastParamOfPath()
    const pattern = new RegExp('[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}')
    if (lastParam && pattern.test(lastParam)) {
      loadCartFromOrder(lastParam)
    }
    if (JSON.stringify(cart) !== JSON.stringify({ cartItems: [] })) {
      verifyAndUpdateCart(cart)
      cartUpdate(cart, skusNotAvailable, apiCalled, discount, this.setCartState, this.setApiCalled, false)
    }
  }

  componentDidUpdate(prevProps) {
    const { cart } = this.props
    const { discount, skusNotAvailable } = this.state
    if (prevProps.cart !== cart) {
      verifyAndUpdateCart(cart)
      cartUpdate(cart, skusNotAvailable, apiCalled, discount, this.setCartState, this.setApiCalled, discount !== null)
    }
  }

  render() {
    const promoTargetSkus = getPromoTargetSkus()
    return <Cart { ...this.props } { ...this.state } promoTargetSkus={ promoTargetSkus } setCartState={ this.setCartState } />
  }
}

const mapStateToProps = state => {
  return { ...state.global, ...state.cart }
}

const mapDispatchToProps = dispatch => {
  return {
    onAddActiveAddon: (productSku, addon, cartIndex) => dispatch(addActiveAddon(productSku, addon, cartIndex)),
    onRemoveActiveAddon: (productSku, addonSku, cartIndex) =>
      dispatch(removeActiveAddon(productSku, addonSku, cartIndex)),
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CartWrapper)
