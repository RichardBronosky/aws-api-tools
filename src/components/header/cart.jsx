import React from 'react'
import { connect } from 'react-redux'
import { getCartTotal } from '../../lib/helpers/cart'
import { currencyFormatUS } from '../../lib/helpers/string-helper'
import { isProductStrikeThrough } from '../../lib/helpers/product'
import RTGLink from '../shared/link'
import RemoveFromCart from '../cart/cart-parts/remove-from-cart'
import { productUrl } from '../../lib/helpers/route'

class CartPopout extends React.Component {
  render() {
    const cart = this.props.cart
    return (
      <>
        <div className="section">
          <div id="cartSubtotal" className="subtotal">
            Subtotal: { currencyFormatUS(getCartTotal(cart)) }
          </div>
        </div>
        { (!cart || (cart && typeof cart.cartItems === 'undefined')) && (
          <>
            <div className="section">
              <div className="message">Your shopping cart is empty</div>
            </div>
            { /* <div className="section">
              <div className="notice">
                <p>
                  If you saved items on a previous visit,{ ' ' }
                  <a href="/">Sign In</a> to access your saved cart.
                  </p>
              </div>
            </div> */ }
          </>
        ) }
        <div className="section sticky">
          <RTGLink
            data={ {
              slug: '/checkout',
              title: 'Checkout',
              category: 'header',
              action: 'cart interaction',
              label: 'checkout',
            } }
          >
            <span className="blue-action-btn checkout">Checkout</span>
          </RTGLink>
          <RTGLink
            data={ {
              slug: '/cart',
              title: 'View Full Cart',
              category: 'header',
              action: 'cart interaction',
              label: 'view full cart',
            } }
            className="bold-arrow-link"
          >
            View Full Cart >
          </RTGLink>
        </div>
        { cart && cart.cartItems && cart.cartItems.length > 0 && (
          <>
            { cart.cartItems &&
              cart.cartItems.map((item, index) => {
                const product = item.product
                const strikethrough = isProductStrikeThrough(product)
                return (
                  <div key={ index } className="line-item grid-x">
                    <div className="product cell small-10">
                      <RTGLink
                        data={ {
                          url: productUrl(product.title, product.sku),
                        } }
                        aria-hidden="true"
                        tabIndex="-1"
                        disabled={ product.category === 'gift-card' }
                      >
                        <div className="prod-img">
                          <img src={ product.primary_image } alt="" role="presentation" />
                        </div>
                      </RTGLink>
                      <div className="prod-details">
                        <RTGLink
                          data={ {
                            url: productUrl(product.title, product.sku),
                          } }
                          aria-describedby={ `mcq_${ index }mcp_${ index }` }
                          disabled={ product.category === 'gift-card' }
                        >
                          <div className="title">{ product.title }</div>
                        </RTGLink>
                        <div id={ 'mcq_' + index } className="qty">
                          Qty: { item.quantity }
                        </div>
                        <div id={ 'mcp_' + index } className="price">
                          { strikethrough && (
                            <p className="strikethrough">{ currencyFormatUS(item.product.list_price) }</p>
                          ) }
                          <p className={ strikethrough ? 'strikethrough-sale' : '' }>{ currencyFormatUS(item.price) }</p>
                        </div>
                      </div>
                    </div>

                    <div className="cell small-2">
                      <RemoveFromCart headerCart product={ product } index={ index } />
                    </div>
                  </div>
                )
              }) }
          </>
        ) }
      </>
    )
  }
}

const mapStateToProps = state => {
  return { ...state.cart }
}

export default connect(
  mapStateToProps,
  null
)(CartPopout)
