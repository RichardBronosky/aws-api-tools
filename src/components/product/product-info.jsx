import React from 'react'
import { connect } from 'react-redux'
import ProductInfoGrid from './views/product-info-grid'
import ProductInfoList from './views/product-info-list'
import { productAvailability, productPrice, productOnSale, isProductStrikeThrough } from '../../lib/helpers/product'
import { productFinancing } from '../../lib/helpers/finance'

class ProductInfo extends React.Component {
  state = {
    currentImage: null,
    addons: null,
    showFinance: false,
    financeAmount: 0,
  }

  componentDidMount() {
    this.setUp(this.props.data)
  }

  componentDidUpdate(prevProps) {
    if (prevProps.data !== this.props.data) {
      this.setUp(this.props.data)
    }
  }

  setUp = product => {
    this.setAddons(product)
    const finance = productFinancing(productPrice(product))
    this.setState({
      financeAmount: finance.financeAmount,
      showFinance: finance.financeAmount >= 10,
    })
  }

  setImage = image => {
    this.setState({
      currentImage: image,
    })
  }

  setAddons = product => {
    const addons = product.addon_items
    if (addons) {
      this.setState({ addons: addons })
    }
  }

  render() {
    const {
      data,
      fullWidth,
      gridWidth,
      viewType,
      index,
      isMobile,
      source,
      displayQuantity,
      promotions,
      orderedProductVariations,
      setProduct,
      last,
    } = this.props
    const { showFinance, financeAmount } = this.state
    const product = data
    const price = productPrice(product)
    const strikethrough = isProductStrikeThrough(product)
    const strikethroughPrice = productPrice(product, true)
    const availability = productAvailability(product)
    const sale = productOnSale(product)
    const variationCount = product && product.variation_types ? Object.keys(product.variation_types).length : 0
    return (
      <>
        { viewType === 'grid' && (
          <ProductInfoGrid
            product={ product }
            fullWidth={ fullWidth }
            price={ price }
            strikethrough={ strikethrough }
            strikethroughPrice={ strikethroughPrice }
            availability={ availability }
            sale={ sale }
            currentImage={ this.state.currentImage }
            setImage={ this.setImage }
            variationCount={ variationCount }
            isMobile={ isMobile }
            index={ index }
            source={ source }
            gridWidth={ gridWidth }
            addons={ this.state.addons }
            displayQuantity={ displayQuantity }
            showFinance={ showFinance }
            financeAmount={ financeAmount }
            setProduct={ setProduct }
            orderedProductVariations={ orderedProductVariations }
            last={ last }
          />
        ) }
        { viewType === 'list' && (
          <ProductInfoList
            product={ product }
            fullWidth={ fullWidth }
            price={ price }
            strikethrough={ strikethrough }
            strikethroughPrice={ strikethroughPrice }
            availability={ availability }
            sale={ sale }
            currentImage={ this.state.currentImage }
            setImage={ this.setImage }
            variationCount={ variationCount }
            isMobile={ isMobile }
            index={ index }
            source={ source }
            gridWidth={ gridWidth }
            addons={ this.state.addons }
            showFinance={ showFinance }
            financeAmount={ financeAmount }
            promotions={ promotions }
          />
        ) }
      </>
    )
  }
}

const mapStateToProps = state => {
  return { ...state.global }
}

export default connect(
  mapStateToProps,
  null
)(ProductInfo)
