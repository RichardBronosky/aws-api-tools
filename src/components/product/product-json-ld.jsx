import React from 'react'
import { productPrice } from '@helpers/product'
import { stripHtml } from '@helpers/string-helper'
import { productUrl } from '@helpers/route'

export default function ProductJsonLd({ product }) {
  const data = `{ 
    "@context": "http://schema.org/",
    "@type": "Product",
    "name": "${ product.title }",
    "image": "${ [].concat(product.primary_image).concat(product.alternate_images) }",
    "description": "${ product.description ? stripHtml(product.description) : '' }",
    "sku": "${ product.sku }",
    "brand": {
      "@type": "Thing",
      "name": "${ product.brand ? product.brand : 'Rooms To Go' }"
    },
    "offers": {
      "@type": "Offer",
      "url": "${ window && window.location.origin }${ productUrl(product.title, product.sku) }",
      "priceCurrency": "USD",
      "price": "${ productPrice(product) }",
      "itemCondition": "http://schema.org/NewCondition",
      "seller": {
        "@type": "Organization",
        "name": "Rooms To Go"
      }
    }
  }`

  return <script type="application/ld+json" dangerouslySetInnerHTML={ { __html: data } } />
}
