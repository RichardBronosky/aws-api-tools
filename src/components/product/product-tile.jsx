import React from 'react'
import { graphql } from 'gatsby'
import { fetchProductBySku } from '../../lib/services/product'
import ProductInfo from './product-info'
import '../../assets/css/components/product/product-tile.sass'
import '../../assets/css/components/product/views/product-info-grid.sass'
import '../../assets/css/components/product/views/product-info-list.sass'

export default class ProductTile extends React.Component {
  state = {
    product: null,
  }

  componentDidMount() {
    let sku
    if (!this.props.data || this.props.sku) {
      sku = this.props.sku
    } else if (this.props.data.sku) {
      sku = this.props.data.sku
    }
    if ((!this.props.data && sku) || this.props.data.contentful_id) {
      fetchProductBySku(sku).then(
        result => this.setState({ isLoaded: true, product: result }),
        error => this.setState({ isLoaded: true, error })
      )
    }
  }

  swapProductInTile = product => {
    this.setState({ product })
  }

  render() {
    const { fullWidth, gridWidth, viewType, source, index, displayQuantity, data, last } = this.props
    /* Pass product variations from props to maintain variation order */
    const orderedProductVariations = data && data.variations ? data.variations : null
    return (
      <>
        { (this.state.product || (data && data.sku)) && (
          <ProductInfo
            data={ this.state.product || data }
            orderedProductVariations={ orderedProductVariations }
            fullWidth={ fullWidth }
            viewType={ viewType ? viewType : 'grid' }
            index={ index }
            source={ source }
            gridWidth={ gridWidth }
            displayQuantity={ displayQuantity }
            setProduct={ this.swapProductInTile }
            last={ last }
          />
        ) }
      </>
    )
  }
}

export const ProductTileFragment = graphql`
  fragment ProductTile on ContentfulProductTile {
    sku
    contentful_id
    __typename
  }
`
