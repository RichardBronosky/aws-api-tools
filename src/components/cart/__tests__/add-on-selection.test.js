import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { store } from '../../../redux/store'
import * as productHelper from '../../../lib/helpers/product'
import * as cartHelper from '../../../lib/helpers/cart'
import { mockProduct1, mockAddonItem, mockActiveAddon } from '../../../lib/mocks/productDataMocks'
import AddonSelection from '../cart-parts/add-on-selection'

describe('AddonSelection', () => {
  const testProps = {
    product: mockProduct1,
    activeAddons: null,
    cartIndex: 0,
    productQuantity: 1,
  }

  it('renders correctly without addons', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <AddonSelection { ...testProps } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with addon items, and without active addons', () => {
    const props = {
      ...testProps,
      product: { ...mockProduct1, addon_items: [mockAddonItem] },
    }
    const tree = renderer
      .create(
        <Provider store={ store }>
          <AddonSelection { ...props } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with addon items, and with active addons', () => {
    const props = {
      ...testProps,
      product: { ...mockProduct1, addon_items: [mockAddonItem] },
      activeAddons: [mockActiveAddon],
    }
    const tree = renderer
      .create(
        <Provider store={ store }>
          <AddonSelection { ...props } />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('calls isActiveAddon when simulating checkbox onChange when addon is not active', () => {
    const tree = renderer.create(<AddonSelection { ...testProps } />)
    productHelper.isActiveAddon = jest.fn()
    cartHelper.removeActiveAddon = jest.fn()
    tree.root.findByProps({ type: 'checkbox' }).props.onChange()
    expect(productHelper.isActiveAddon).toHaveBeenCalled()
    expect(cartHelper.removeActiveAddon).not.toHaveBeenCalled()
  })

  it('calls isActiveAddon when simulating checkbox onChange when addon is active', () => {
    const tree = renderer.create(<AddonSelection { ...{ ...testProps, activeAddons: [mockActiveAddon] } } />)
    productHelper.isActiveAddon = jest.fn()
    cartHelper.fetchAndAddActiveAddon = jest.fn()
    tree.root.findByProps({ type: 'checkbox' }).props.onChange()
    expect(productHelper.isActiveAddon).toHaveBeenCalled()
    expect(cartHelper.fetchAndAddActiveAddon).toHaveBeenCalled()
  })
})
