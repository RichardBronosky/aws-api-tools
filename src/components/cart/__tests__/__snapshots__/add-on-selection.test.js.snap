// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddonSelection renders correctly with addon items, and with active addons 1`] = `
<div
  className="grid-x cart-addons small-12 medium-6 large-8 large-offset-4"
>
  <div
    className="addon-header add small-12 "
  >
    Add:
  </div>
  <ul
    className="addon-group grid-x"
  >
    <div
      className="small-12 addon-row"
    >
      <span>
        Twin Bunkie Boards
         (
        2
        )
         
        $99.98
      </span>
      <input
        checked={true}
        name="Twin Bunkie Boards"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </ul>
</div>
`;

exports[`AddonSelection renders correctly with addon items, and without active addons 1`] = `
<div
  className="grid-x cart-addons small-12 medium-6 large-8 large-offset-4"
>
  <div
    className="addon-header add small-12 "
  >
    Add:
  </div>
  <ul
    className="addon-group grid-x"
  >
    <div
      className="small-12 addon-row"
    >
      <span>
        Twin Bunkie Boards
         (
        2
        )
         
        $99.98
      </span>
      <input
        checked={false}
        name="Twin Bunkie Boards"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </ul>
</div>
`;

exports[`AddonSelection renders correctly without addons 1`] = `
<div
  className="grid-x cart-addons small-12 medium-6 large-8 large-offset-4"
>
  <div
    className="addon-required"
  >
    <p>
      Twin Bunkie Boards
       (
      2
      ) 
      <span
        className="addon-declined"
      >
        Declined
      </span>
    </p>
    <button
      className="bunkie-board addon-learn-more"
      gtm-action="click"
      gtm-category="bunkie-board-modal"
      gtm-label="cart"
      onClick={[Function]}
    >
      What is a bunkie board? â“˜
    </button>
  </div>
  <div
    className="addon-header add small-12 "
  >
    Add:
  </div>
  <ul
    className="addon-group grid-x"
  >
    <div
      className="small-12 addon-row"
    >
      <span>
        Twin Bunkie Boards
         (
        2
        )
         
        $99.98
      </span>
      <input
        checked={false}
        name="Twin Bunkie Boards"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </ul>
</div>
`;
